src/java.base/share/classes/java/lang/invoke/MethodHandleNatives.java

```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2008, 2023, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2008, 2024, Oracle and/or its affiliates. All rights reserved.
```

Comment:
Rather than patch `LambdaFormEditor` and `MethodTypeForm`, I think we should instead (temporarily?) change the field `USE_SOFT_CACHE` here:
```java
    static final boolean USE_SOFT_CACHE = true;
```
OpenJ9 doesn't need to support "dumping the AOTCache" (which might require the value to be `false`).

Please also retain the import statements added by [8331497: Implement JEP 483: Ahead-of-Time Class Loading & Linking](https://github.com/ibmruntimes/openj9-openjdk-jdk/commit/be095e78c3f87a3a6ef0746d0ec55bf14ff55e5c).

Reply:
> static final boolean USE_SOFT_CACHE = true;

If this is hardcoded as `true`, the system property `java.lang.invoke.MethodHandleNatives.USE_SOFT_CACHE` won't be able to change JVM `AOTCache` behaviour.

> OpenJ9 doesn't need to support "dumping the AOTCache" (which might require the value to be false).

OpenJ9 uses OpenJDK MethodHandle implementation since JDK17 and gets the "dumping the AOTCache" function. I don't see any OpenJ9-specific support is required.

Reply:
I couldn't find anything in this repository, or elsewhere, that would set `java.lang.invoke.MethodHandleNatives.USE_SOFT_CACHE=false`, so I don't see any need to query that system property. It's not clear to me whether you're agreeing with me nor not.

Reply:
This is a system property like `java.lang.invoke.MethodHandle.DEBUG_xxx or TRACE_xxx` that can be set by an application at the command line.
These properties are useful for runtime controlling the application behaviours.

Reply:
Yes, in general, system properties can affect behavior, but in this case it seems the property is defined by a Hotspot JVM based on `-XX:AOTCache=xxx` command-line option. OpenJ9 ignores such options, and I found no place that uses `-Djava.lang.invoke.MethodHandleNatives.USE_SOFT_CACHE=false` so there's nothing to be gained by trying to follow that system property.

Reply:
> Please also retain the import statements added by [8331497: Implement JEP 483: Ahead-of-Time Class Loading & Linking](https://github.com/ibmruntimes/openj9-openjdk-jdk/commit/be095e78c3f87a3a6ef0746d0ec55bf14ff55e5c).

`LambdaFormEditor.java` and `MethodTypeForm.java` should be identical to upstream.

Reply:
Hmm, restored the upstream version.